x-pack/plugins/security_solution/server/assistant/tools/alert_counts/alert_counts_tool.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SearchResponse } from '@elastic/elasticsearch/lib/api/types';
+import { DynamicTool } from 'langchain/tools';
+
+import { requestHasRequiredAnonymizationParams } from '@kbn/elastic-assistant-plugin/server/lib/langchain/helpers';
+import type { AssistantTool, AssistantToolParams } from '@kbn/elastic-assistant-plugin/server';
+import { getAlertsCountQuery } from './get_alert_counts_query';
+import { APP_ID } from '../../../../common';
+
+export interface AlertCountsToolParams extends AssistantToolParams {
+  alertsIndexPattern: string;
+}
+export const ALERT_COUNTS_TOOL_DESCRIPTION =
+  'Call this for the counts of last 24 hours of open alerts in the environment, grouped by their severity';
+
+export const ALERT_COUNTS_TOOL: AssistantTool = {
+  id: 'alert-counts-tool',
+  name: 'AlertCountsTool',
+  description: ALERT_COUNTS_TOOL_DESCRIPTION,
+  sourceRegister: APP_ID,
+  isSupported: (params: AssistantToolParams): params is AlertCountsToolParams => {
+    const { request, alertsIndexPattern } = params;
+    return requestHasRequiredAnonymizationParams(request) && alertsIndexPattern != null;
+  },
+  getTool(params: AssistantToolParams) {
+    if (!this.isSupported(params)) return null;
+    const { alertsIndexPattern, esClient } = params as AlertCountsToolParams;
```

Comment:
Not required for this PR, but per our offline discussion the cast feels strange here because `isSupported` is a type guard that validates `params`. (This comment is applicable to all implementations of `getTool` in the PR.)

Reply:
I gave another pass at this and was not successful in getting type narrowing working with the `isSupported` function. Tried hoisting the function, re-declaring, and still needing the `as` cast. Will try again with fresh 👀's tomorrow.